### PR TITLE
added test cases for parseString()

### PR DESCRIPTION
**Pull request by rperez4**
Background: Before updating the components in the Dagger Browser, I wanted to integrate unit testing to test the parseString() method I created that will shorten the name of each node in the Dagger Graph to help readability. Testing will help make sure the logic fo the parseString() method is sound and ready to integrate in the next PR

Change: Added two methods in the browser/service/NodeAutosuggestService class that will help shorten the name of each node. Added 5 unit tests in the browser/__tests__/NodeAutosuggestService.test.tsx file.

Test Plan: Use various node names from the Dagger Graph to ensure proper testing.

### CODE REVIEW

File: browser/src/service/NodeAutosuggestService.tsx [hunk 1]
```diff
@@ -1,15 +1,71 @@
...
+export class NodeAutosuggestService {
+    //constructor is empty, only necessary to access methods parseString, and addQuestionsMarks
+    constructor() {}
+
+    /* 
+    Component is passed in and a shorter name of the component is returned, 
+    Time Complexity O(n)
+    Space Complexity O(n)
+    */
+    parseString(component: string) {
```
Comment (bjdodson) [inline browser/src/service/NodeAutosuggestService.tsx:12]: This isn't a generic "string parser" -- we're parsing a Dagger binding key. Consider `displayNameForKey()` or similar.

Comment (bjdodson) [inline browser/src/service/NodeAutosuggestService.tsx:12]: add return type

File: browser/src/service/NodeAutosuggestService.tsx [hunk 2]
```diff
@@ -1,15 +1,69 @@
...
-    
-    add(a: number, b: number){
-        return a+b
+export class NodeAutosuggestService {
+    //constructor is empty, only necessary to access methods parseString, and addQuestionsMarks
+    constructor() {}
+
+    /* 
+    Component is passed in and a shorter name of the component is returned, 
+    Time Complexity O(n)
```
Comment (bjdodson) [inline browser/src/service/NodeAutosuggestService.tsx:9]: remove complexities (not technically accurate since there are nested regexp)

File: browser/src/service/NodeAutosuggestService.tsx (hunk 1 above)
Comment (bjdodson) [reply browser/src/service/NodeAutosuggestService.tsx:12]: Also move this to a helper/util file, since it doesn't relate to "NoteAutoSuggestService" and will be used in several places.

File: browser/src/__tests__/NodeAutosuggest.test.tsx [hunk 3]
```diff
@@ -1,22 +1,49 @@
...
+        "com.snap.durablejob.DurableJobProcessor<?,?>>> com.snap.lenses.app.LensesDurableJobComponent_MultibindingsModule"+
+        "#provide_socialUnlockResponseCacheCleanupProcessorMultibinding")).equals(
+           "Map<Class<DurableJob<?>> Provider<DurableJobProcessor<?>>> LensesDurableJobComponentMultibindingsModule"
+          )
+    })
+
+    test('test 4', () => {
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString("@io.plaidapp.core.dagger.DesignerNewsApi okhttp3.OkHttpClient")).equals(
+        "DesignerNewsApiOkHttpClient")
```
Comment (bjdodson) [inline browser/src/__tests__/NodeAutosuggest.test.tsx:40]: ```@DesignerNewsApiOkHttpClient OkHttpClient```

File: browser/src/__tests__/NodeAutosuggest.test.tsx [hunk 4]
```diff
@@ -1,22 +1,49 @@
...
 
-    test('sum 1 + 2 = 3' , () => {
-      const nodeContainer = shallow(<NodeAutoSuggestService/>);
-      expect((nodeContainer.instance() as NodeAutoSuggestService).add(1,2)).equals(3)
+    test('test 1', () =>{
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString(
+        "java.util.Map<java.lang.Class<? extends com.snap.durablejob.DurableJob<?>>,javax.inject.Provider<"+
+        "com.snap.durablejob.DurableJobProcessor<?,?>>>")).equals(
+        "Map<Class<DurableJob<?>> Provider<DurableJobProcessor<?>>>"
```
Comment (bjdodson) [inline browser/src/__tests__/NodeAutosuggest.test.tsx:17]: ```Map<Class<DurableJob<?>>, Provider<DurableJobProcessor<?>>>```

File: browser/src/__tests__/NodeAutosuggest.test.tsx [hunk 5]
```diff
@@ -1,22 +1,49 @@
...
+    test('test 2', () => {
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString("com.snap.dagger.browser.sample.CoffeeApp.CoffeeShop")).equals(
+        "CoffeeApp.CoffeeShop")
+    })
+
+    test('test 3', () => {
+      const nodeAutoSuggest = new NodeAutoSuggestService();
+      expect(nodeAutoSuggest.parseString(
+        "java.util.Map<java.lang.Class<? extends com.snap.durablejob.DurableJob<?>>,javax.inject.Provider<"+
```
Comment (bjdodson) [inline browser/src/__tests__/NodeAutosuggest.test.tsx:30]: Is this input an actual key from ComponentManifests.json?

File: browser/src/service/NodeAutosuggestService.tsx [hunk 6]
```diff
@@ -1,15 +1,69 @@
...
+    Space Complexity O(n)
+    */
+    parseString(component: string) {
+        var shorter_component = ""
+        for (var i = 0; i < component.length; i++) {
```
Comment (bjdodson) [inline browser/src/service/NodeAutosuggestService.tsx:14]: this is pretty hard to follow-- could it be more token-based,  eg given your current processing state, are you looking for a qualifier (start of input, starts with `@`,  a general type name,  a type parameter/generic (`<>`), type parameter bounds (`?` inside of type parameter)

File: browser/src/__tests__/NodeAutosuggest.test.tsx (hunk 3 above)
Comment (bjdodson) [reply browser/src/__tests__/NodeAutosuggest.test.tsx:40]: here, we find the qualifier,  convert its type to a simple display name,  and then add it to the beginning of the result as `@[simpleName]`

File: browser/src/__tests__/NodeAutosuggest.test.tsx (hunk 5 above)
Comment (rperez4) [reply browser/src/__tests__/NodeAutosuggest.test.tsx:30]: https://ci-log-proxy.gae.sc-corp.net/log-proxy/snapengine-builder-artifacts/android-dagger-browser/latest/index.html#/com.snap.mushroom.dagger.durablejob.MushroomFeatureGraphDurableJobRegistry/node/java.util.Map%3Cjava.lang.Class%3C%3F%20extends%20com.snap.durablejob.DurableJob%3C%3F%3E%3E%2Cjavax.inject.Provider%3Ccom.snap.durablejob.DurableJobProcessor%3C%3F%2C%3F%3E%3E%3E

I got this dependency from this link you sent sometime ago

File: browser/src/__tests__/NodeAutosuggest.test.tsx [hunk 7]
```diff
@@ -1,22 +1,49 @@
...
 
 describe("Map of class to Provider has correct short name", () => {
-    test("should display a blank login form, with remember me checked by default", async () => {
-      // ???
-    });
 
-    test('sum 1 + 2 = 3' , () => {
-      const nodeContainer = shallow(<NodeAutoSuggestService/>);
-      expect((nodeContainer.instance() as NodeAutoSuggestService).add(1,2)).equals(3)
+    test('test 1', () =>{
```
Comment (weixc911214) [inline browser/src/__tests__/NodeAutosuggest.test.tsx:12]: maybe give the test a better name than `test 1`

File: browser/src/service/NodeAutosuggestService.tsx (hunk 6 above)
Comment (weixc911214) [reply browser/src/service/NodeAutosuggestService.tsx:14]: Agree, the logic here is more like ACM-style, considering making it more readable and it's easier for others to maintain in the future.

**Review by weixc911214 — COMMENTED** [review]
Left some comments, also for the **Test** section in the description for this PR, do you mind add the test command you run locally so others can reproduce it in the future.

File: browser/src/__tests__/NameForKey.test.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+describe("Map of class to Provider has correct short name", () => {
+
+    const nodeAutoSuggest : Helper = new Helper()
+  
+    test('test 1, from CoffeeApp Example', () => {
+      expect(nodeAutoSuggest.displayNameForKey("com.snap.dagger.browser.sample.CoffeeApp.CoffeeShop")).equals(
+        "CoffeeApp.CoffeeShop")
+    })
+
+    test('test 3, from HomeComponent', () => {
```
Comment (weixc911214) [inline browser/src/__tests__/NameForKey.test.tsx:17]: `from HomeComponent` is not a good test name, maybe something like `displayNameForKey  should return simplified annotation and class name`. Not sure about the best practice for test naming for typescript, but you got what I mean.

File: browser/src/__tests__/NameForKey.test.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      expect(nodeAutoSuggest.displayNameForKey("@io.plaidapp.core.dagger.DesignerNewsApi okhttp3.OkHttpClient")).equals(
+        "@DesignerNewsApi OkHttpClient")
+    })
+
+    test('test 4, from DribleSearchConverter.Factory', () => { 
+      expect(nodeAutoSuggest.displayNameForKey("io.plaidapp.core.dribbble.data.search.DribbbleSearchConverter.Factory")).equals(
+        "DribbbleSearchConverter.Factory")
+    })
+
+    test('test 5, from HomeComponent', () => {
```
Comment (weixc911214) [inline browser/src/__tests__/NameForKey.test.tsx:27]: test 5 is similar to test 1, you can remove one of these 2 and rename the other one like `displayNameForKey should return simple name for fully qualified name`

File: browser/src/__tests__/NameForKey.test.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+      expect(nodeAutoSuggest.displayNameForKey("io.plaidapp.core.designernews.data.login.LoginLocalDataSource")).equals("LoginLocalDataSource")
+    })
+
+    test('test 6, from MushroomFeatureGraphDurableJobRegistry', () => {
```
Comment (weixc911214) [inline browser/src/__tests__/NameForKey.test.tsx:31]: rename test to something like `displayNameForKey should return simple name for class names in Map`

File: browser/src/util/NameForKey.tsx [hunk 11]
```diff
@@ -0,0 +1,72 @@
+export class Helper {
+    //constructor is empty, only necessary to access methods parseString, and addQuestionsMarks
+    constructor() {}
+
+    /* 
+    Component is passed in and a shorter name of the component is returned,
```
Comment (weixc911214) [inline browser/src/util/NameForKey.tsx:6]: the doc seems not finished.

**Review by weixc911214 — APPROVED** [review]

File: browser/src/util/NameForKey.tsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
+export class Helper {
```
Comment (bjdodson) [inline browser/src/util/NameForKey.tsx:1]: Let's call this `DisplayNameHelper`

**Review by bjdodson — APPROVED** [review]
